tuist/tuist
# Reuse binaries for tuist test, generate test project in the current working directory

Resolves https://github.com/tuist/tuist/issues/5281
Resolves https://github.com/tuist/tuist/issues/5184
Resolves https://github.com/tuist/tuist/issues/5034

### Short description 📝

This PR does two things:
- `tuist test` generates the project in-place. That resolves issues #5281 and #5184. Generating project in the custom `~/.tuist/Cache/Projects` directory was causing confusion. While we will now re-generate projects for the purposes of `tuist test`, this can actually be a plus. If tests fail, developers can easily open the generated project and retest their code using the Xcode UI while having the project lean.
- `tuist test` will now reuse the binaries generated by `tuist cache warm` which will make the tests a lot faster as developers and the CI will no longer need to rebuild modules that have been pre-warmed. **Note:** This functionality is available only to users of Tuist Cloud.

### How to test the changes locally 🧐

Run `tuist test` for a project that has tests and Tuist Cloud. You can use `tuist/tuist` as well. The generated project should use binaries and only unit tests should be integrated as sources.

### Contributor checklist ✅


### Reviewer checklist ✅
